host/src/request.rs

```diff
@@ -1,77 +1,289 @@
+    /// transaction.
+    pub l1_rpc: Option<String>,
+    #[arg(long, require_equals = true)]
+    /// The beacon node URL for retreiving data blobs.
+    pub beacon_rpc: Option<String>,
+    #[arg(long, require_equals = true)]
+    /// The network to generate the proof for.
+    pub network: Option<String>,
+    #[arg(long, require_equals = true)]
+    // Graffiti.
+    pub graffiti: Option<String>,
+    #[arg(long, require_equals = true)]
+    /// The protocol instance data.
+    pub prover: Option<String>,
+    #[arg(long, require_equals = true)]
+    /// The proof type.
+    pub proof_type: Option<String>,
+    #[command(flatten)]
+    /// Any additional prover params in JSON format.
+    pub prover_args: ProverSpecificOpts,
 }
 
-#[derive(Debug, Serialize, Deserialize)]
-pub struct JsonRpcError {
-    pub code: i32,
-    pub message: String,
+#[derive(Default, Clone, Serialize, Deserialize, Debug, ToSchema, Args)]
+pub struct ProverSpecificOpts {
+    pub native: Option<Value>,
+    pub sgx: Option<Value>,
+    pub sp1: Option<Value>,
+    pub risc0: Option<Value>,
 }
 
-#[derive(Debug, Serialize)]
-pub struct JsonRpcResponseError {
-    pub jsonrpc: String,
-    pub id: serde_json::Value,
-    pub error: JsonRpcError,
+impl From<ProverSpecificOpts> for HashMap<String, Value> {
+    fn from(value: ProverSpecificOpts) -> Self {
+        HashMap::from_iter(
+            [
+                ("native", value.native.clone()),
+                ("sgx", value.sgx.clone()),
+                ("sp1", value.sp1.clone()),
+                ("risc0", value.risc0.clone()),
+            ]
+            .into_iter()
+            .filter_map(|(name, value)| value.map(|v| (name.to_string(), v))),
+        )
+    }
 }
 
-#[derive(Debug, Serialize, Deserialize)]
-pub struct JsonRpcResponse<T> {
-    pub jsonrpc: String,
-    pub id: serde_json::Value,
-    pub result: Option<T>,
+impl ProofRequestOpt {
+    /// Read a partial proof request config from a file.
+    pub fn from_file<T>(path: T) -> Result<Self, HostError>
+    where
+        T: AsRef<Path>,
+    {
+        let file = std::fs::File::open(path)?;
+        let reader = std::io::BufReader::new(file);
+        let config: Value = serde_json::from_reader(reader)?;
+        Self::deserialize(&config).map_err(|e| e.into())
+    }
+
+    /// Merge a partial proof request into current one.
+    pub fn merge(&mut self, other: &Value) -> Result<(), HostError> {
+        let mut this = serde_json::to_value(&self)?;
+        merge(&mut this, other);
+        *self = serde_json::from_value(this)?;
+        Ok(())
+    }
 }
 
-#[derive(Debug, Serialize, Deserialize)]
-pub struct JsonRpcRequest<T: Serialize> {
-    pub jsonrpc: String,
-    pub id: serde_json::Value,
-    pub method: String,
-    pub params: T,
+impl TryFrom<ProofRequestOpt> for ProofRequest {
+    type Error = HostError;
+
+    fn try_from(value: ProofRequestOpt) -> Result<Self, Self::Error> {
```

Comment:
If I comment out the network for example in the JSON-RPC, I get "missing network", but I still have the network defined in the config file. I think this is unexpected because I would expect the network defined in the config file to be used when it's not overridden by other inputs.